docs/source/starter/converting.rst

```diff
@@ -76,38 +85,121 @@ To add an (optional) validation loop add logic to the
 
 The test loop will not be used until you call.
 
-.. code-block::
+.. code-block:: python
 
     trainer.test()
 
-.. tip:: ``.test()`` loads the best checkpoint automatically
+.. tip:: ``trainer.test()`` loads the best checkpoint automatically by default if checkpointing is enabled.
+
+--------
+
+*******************************
+6. Find the Predict Loop "meat"
+*******************************
+
+To add an (optional) prediction loop add logic to the
+:meth:`~pytorch_lightning.core.lightning.LightningModule.predict_step` hook (make sure to use the hook parameters, ``batch`` and ``batch_idx`` in this case).
+If you don't override ``predict_step`` hook, it by default calls ``forward`` method on the batch.
+
+.. testcode::
+
+    class LitModel(LightningModule):
+        def predict_step(self, batch, batch_idx):
+            x, y = batch
+            pred = self(x)
+            return pred
+
+.. note:: ``model.eval()`` and ``torch.no_grad()`` are called automatically for prediction.
+
+The predict loop will not be used until you call.
+
+.. code-block:: python
+
+    trainer.predict()
+
+.. tip:: ``trainer.predict()`` loads the best checkpoint automatically by default if checkpointing is enabled.
+
+--------
+
+******************************************
+7. Remove any .cuda() or .to(device) Calls
+******************************************
+
+Your :doc:`LightningModule <../common/lightning_module>` can automatically run on any hardware!
+
+If you have any explicit calls to ``.cuda()`` or ``.to(device)``, you can remove them since Lightning makes sure that the data coming from :class:`~torch.utils.data.DataLoader`
+and all the :class:`~torch.nn.Module` instances initialized inside ``LightningModule.__init__`` are moved to the respective devices automatically.
+
+.. testcode::
+
+    class LitModel(LightningModule):
+        def __init__(self):
+            super().__init__()
+            self.register_buffer("running_mean", torch.zeros(num_features))
+
+If you still need to access the current device, you can use ``self.device`` anywhere in ``LightningModule`` except ``__init__`` method. You are initializing a
+:class:`~torch.Tensor` within ``LightningModule.__init__`` method and want it to be moved to the device automatically you must :meth:`~torch.nn.Module.register_buffer`
+to register it as a parameter.
+
+.. testcode::
+
+    class LitModel(LightningModule):
+        def training_step(self, batch, batch_idx):
+            z = torch.randn(4, 5, device=self.device)
+            ...
 
 --------
 
-6. Remove any .cuda() or to.device() calls
-==========================================
-Your :doc:`lightning module <../common/lightning_module>` can automatically run on any hardware!
+**************
+8. Plugin Data
+**************
+
+To plugin your DataLoaders, you can override the respective dataloader hooks:
+
+.. testcode::
+
+    class LitModel(LightningModule):
```

Comment:
I thought this is not recommended way...

Reply:
what do you mean?